lib/network_monitor.c

```diff
@@ -102,15 +106,49 @@ struct mptcpd_rtm_addr
         void const *const addr;
 };
 
+/**
+ * @struct mptcpd_addr_info
+ *
+ * @brief Convenience structure to bundle address information.
+ *
+ * Note that Network monitor users (tests and sspi plugin) assume
+ * that interface->addr is a list of sockaddr. Placing the
+ * sockaddr_storage as the first field allow expanding the address
+ * metadata without breaking that assumption
+ */
+struct mptcpd_addr_info
```

Comment:
Perhaps in another pull request, we should probably rename this private structure since its name collides with the `struct mptcpd_addr_info` in `<mptcpd/addr_info.h>`.  This is something I missed when I introduced the latter.

Reply:
I agree a rename would help. I suggest to use a different pull req, to avoid growing this one too much